<?xml version="1.0"?>

<launch>

  <arg name="save" default="false" doc="Set to true if you wanted save bag file of trial"/>

  <arg name="subject_id" default="test" doc="Unique string identifier for the subject"/>
  <arg name="assistance_block" default="no" doc="Assistance type for the testing block. For loading up the appropriate trial block file"/>
  <arg name="block_id" default="0" doc="block_id for the assistance type for the testing block"/>
  <arg name="training" default="0" doc="bool string indicating whether the simulator is run in training mode or not"/>
  <!-- record bags -->
  <!-- bag file name parameters -->
  <group if="$(arg save)">
    <include file="$(find simulators)/launch/record.launch">
      <arg name="bag_prefix" value="$(arg subject_id)_$(arg assistance_block)_assistance_$(arg block_id)_mode_inference"/>
    </include>
  </group>

  
	<arg name="SNP" default="false" doc="Set to true if using sip/puff to control arm."/>
  <arg name="JOINT" default="false" doc="Set to true if joint velocity control"/>
  
  <node pkg="simulators" output="screen" type="continuous_world_SE2_simulator.py" name="continuous_world_SE2_simulator"  args="$(arg subject_id) $(arg assistance_block) $(arg block_id) $(arg training)"/>
  <node pkg="inference_and_correction" output="screen" type="goal_inference_and_correction.py" name="goal_inference_and_correction"  args="$(arg subject_id)"/>
  
  <include file="$(find teleop_nodes)/launch/teleop_nodes.launch">
    <arg name="SNP" value="$(arg SNP)"/>
    <arg name="JOINT" value="$(arg JOINT)"/>
    <arg name="robot_dim" value="3"/>
    <arg name="finger_dim" value="0"/>
  </include>
  
  

</launch>
